packages/discovery-react-components/src/components/SearchFacets/SearchFacets.tsx

```diff
@@ -183,6 +195,7 @@ const SearchFacets: FC<SearchFacetsProps> = ({
         {hasSelection && (
           <Button
             className={`${settings.prefix}--search-facets__button-clear-all`}
+            id={'search-facets-button-clear-all'}
```

Comment:
same here with identifiers, although this may be unique, nothing excludes a user from doing multiple `<SearchFacets>` on a page. adding an `id` would make it invalid HTML when multiple of the same id reside on the page. you can use `data-testid` or something similar if this is being used for testing

Reply:
@maniax89 for this case and the one mentioned above, would a `data-testid` be suitable for tracking purposes?  With the scenario you mentioned with multiple `SearchFacets` being called, tracking would still only be concerned if that type of button is clicked rather than which specific one.  Since tracking does not record which specific UI element triggered the event.  If not, what would be a suitable method for placing identifiers for these components?

Reply:
for any of the carbon components, i would just make sure it is prefixed with something that is user-controlled if necessary. you can have a top-level attribute like `clearAllButtonId` that defaults to `search-facets-button-clear-all`. that way when users consume it in their applications, they don't experience accessibility violations due to duplicate ids https://www.w3.org/TR/WCAG20-TECHS/failures#F77

Reply:
So user-defined would be something like what is occurring in the `SearchInput` component?
https://github.com/watson-developer-cloud/discovery-components/blob/2f68225950583370a2f224066faba1e1feaf0ea7/packages/discovery-react-components/src/components/SearchInput/SearchInput.tsx#L30